frontend/public/src/components/ReceiptPageDetailCard.js

```diff
@@ -205,9 +207,11 @@ export class ReceiptPageDetailCard extends React.Component<Props> {
                     <td>
                       <div>${line.price}</div>
                     </td>
-                    <td>
-                      <div>${discountAmountText}</div>
-                    </td>
+                    {discountAmountText !== null ? (
+                      <td>
+                        <div>${discountAmountText}</div>
```

Comment:
Small thing: remove the `$` here and move it to the appropriate spots in the switch statement starting on line 28. (it displays "$35% off" for a 35% discount.)